src/arch/riscv64/CodeGen.zig

```diff
@@ -1812,8 +1812,15 @@ fn finishAirBookkeeping(func: *Func) void {
 fn finishAirResult(func: *Func, inst: Air.Inst.Index, result: MCValue) void {
     if (func.liveness.isUnused(inst)) switch (result) {
         .none, .dead, .unreach => {},
-        else => unreachable, // Why didn't the result die?
+        // Why didn't the result die?
+        .register => |r| if (r != .zero) unreachable,
```

Comment:
I don't see why this logic is needed. atomicLoad calls this with `unreach` when the result is unused, which is correct. I disagree that calling finish with zero is valid.

Reply:
> atomicLoad calls this with `unreach` when the result is unused

That's a leftover that I hadn't gotten rid of.

My intent here is that an AIR handler function for a side-effecting instruction should be able to just pick `zero` as the destination register if the instruction is unused, and that's all that needs to be done. If `zero` is not permitted as the result register in `finishAirResult()` for the unused case, then every such handler function needs to do two things: Pick the `zero` register as the destination register *and* remember to pass `unreach`.

Just handling assignment of the result to `zero` in this way seems like the less error-prone option.

Reply:
Hmm, an interesting idea. I see what you're getting at, I'm fine trying this out.

Reply:
Removed the leftover code, the benefit should be a bit clearer now.